src/VirtualClient/VirtualClient.Actions.UnitTests/HPLinpack/HPLinpackExecutorTests.cs

```diff
@@ -0,0 +1,212 @@
+    using VirtualClient.Common.Telemetry;
+
+    [TestFixture]
+    [Category("Unit")]
+    public class HPLinpackExecutorTests
+    {
+        private MockFixture fixture;
+        private DependencyPath mockPath;
+        private DependencyPath currentDirectoryPath;
+
+        private string resultsPath;
+        private string rawString;
+
+        [SetUp]
+        public void SetUpTests()
+        {
+            this.fixture = new MockFixture();
+            this.mockPath = this.fixture.Create<DependencyPath>();
+        }
+
+        [Test]
+        [TestCase(PlatformID.Unix, Architecture.X64)]
+        [TestCase(PlatformID.Unix, Architecture.Arm64)]
+        public async Task HPLExecutorInitializesItsDependenciesAsExpected(PlatformID platform, Architecture architecture)
+        {
+            this.SetupDefaultMockBehavior(platform, architecture);
+            using (TestHPLExecutor executor = new TestHPLExecutor(this.fixture))
+            {
+                this.fixture.ProcessManager.OnCreateProcess = (command, arguments, workingDirectory) =>
+                {
+                    return this.fixture.Process;
+                };
+
+                await executor.InitializeAsync(EventContext.None, CancellationToken.None)
+                    .ConfigureAwait(false);
+
+                string workloadExpectedPath = this.fixture.PlatformSpecifics.Combine(
+                    this.fixture.PlatformSpecifics.GetPackagePath(), $"hpl-{this.fixture.Parameters["HPLVersion"]}");
+
+                Assert.AreEqual(workloadExpectedPath, executor.HPLDirectory);
+            }
+        }
+
+        [Test]
+        [TestCase(PlatformID.Unix, Architecture.Arm64)]
+        public async Task HPLExecutorExecutesWorkloadAsExpectedOnUbuntuArmPlatform(PlatformID platform, Architecture architecture)
+        {
+            this.SetupDefaultMockBehavior(platform, architecture);
+            List<string> expectedCommandsOnLinuxarm64 = new List<string>()
+            {
+                $"sudo ./arm-performance-libraries_22.1_Ubuntu-20.04.sh -a",
+                $"wget http://www.netlib.org/benchmark/hpl/hpl-{this.fixture.Parameters["HPLVersion"]}.tar.gz -O {this.fixture.Parameters["PackageName"]}.tar.gz",
+                $"tar -zxvf {this.fixture.Parameters["PackageName"]}.tar.gz",
+                $"sudo bash -c \"source make_generic\"",
+                $"make arch=Linux_GCC",
+                $"sudo runuser -u azureuser -- mpirun -np {Environment.ProcessorCount} ./xhpl"
+            };
+
+            using (TestHPLExecutor executor = new TestHPLExecutor(this.fixture))
+            {
+                int executedArmCommands = 0;
+
+                this.fixture.ProcessManager.OnCreateProcess = (command, arguments, workingDirectory) =>
+                {
+                    if (expectedCommandsOnLinuxarm64.Any(c => c == $"{command} {arguments}"))
+                    {
+                        executedArmCommands++;
```

Comment:
This is not a definitive way to check for the expected commands to have been executed. A more precise technique is to simply remove the commands from the set of expectedCommands and then Assert.IsEmpty(expectedCommands) at the end.

In fact, the most precise way that validates the expected commands were executed and in the exact order expected, you remove from expectedCommands[0] ONLY each time in the OnCreateProcess.